pkg/cluster/cluster.go

```diff
@@ -37,6 +37,9 @@ import (
 
 // Cluster provides various methods to interact with a cluster.
 type Cluster interface {
+	// ID returns a unique identifier for the current cluster.
+	ID() string
```

Comment:
It should be possible to also set this, what we do currently seems ok as default but whenever there is a controller that interacts with more than one cluster, something more human-readable seems useful

Reply:
Agreed! Could be an option in the builder?